pkg/controller/ps/upgrade.go

```diff
@@ -92,14 +93,17 @@ func (r *PerconaServerMySQLReconciler) smartUpdate(ctx context.Context, sts *app
 			continue
 		}
 
-		log.Info("apply changes to secondary pod", "pod name", pod.Name)
+		log.Info("apply changes to secondary pod", "podName", pod.Name)
 
 		if pod.ObjectMeta.Labels[controllerRevisionHash] == sts.Status.UpdateRevision {
-			log.Info("pod updated updated", "pod name", pod.Name)
+			log.Info("pod updated updated", "podName", pod.Name)
 			continue
 		}
 
-		return r.Client.Delete(ctx, &pod)
+		err = r.Client.Delete(ctx, &pod)
+		// We need to wait to allow sts.Status.ReadyReplicas to be updated after pod deletion
+		time.Sleep(5 * time.Second)
```

Comment:
rather than sleeping can we wait until its updated?

Reply:
Yes, I don't like this sleep as well, but it's a bit tricky to wait for it, tried it, but I'll spend a bit more time on it to come up with something better hopefully.

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/405/commits/25c323d23abe58c012a75b926ecaaafd15b39386